### DISCUSSION TITLE
Create a home page

### DISCUSSION THREAD

**Discussion by GaetanRdn** [stackblitz/tutorialkit#203] [Ideas]
It could be great to create a real homepage. It's not really a priority, actually I use a lesson page.

**Comment by AriPerkkio** [comment]
You can do this by adding `defaultRoutes: "tutorial-only"` in your `astro.config.ts`:

```ts
import tutorialkit from "@tutorialkit/astro";
import { defineConfig } from "astro/config";

export default defineConfig({
  integrations: [
    tutorialkit({
      defaultRoutes: "tutorial-only",
    }),
  ],
});
```

Then create a `src/pages/index.astro`. That will be served from `/`.

This was implemented in https://github.com/stackblitz/tutorialkit/pull/93 but is not documented yet.

**Reply by GaetanRdn** [reply]
It works but it does not take the styles of the application.

Maybe because I'm a newbie with Astro 🤷🏼‍♂️.

But if I do only what you say there is no styles in the home page.

**Reply by Nemikolh** [reply]
@GaetanRdn, When you mean no styles, what do you mean exactly? We use UnoCSS which is an atomic css engine: https://unocss.dev/.

Its included via its Astro integration which if I'm not mistaken should always include the generated stylesheet on every pages. In particular you can check that we do not include that one manually by looking in those files:
 - [`[...slug].astro`](https://github.com/stackblitz/tutorialkit/blob/main/packages/astro/src/default/pages/%5B...slug%5D.astro)
 - [`Layout.astro`](https://github.com/stackblitz/tutorialkit/blob/main/packages/astro/src/default/layouts/Layout.astro)

If it doesn't seem to be working you can import it explicetly by adding `import "uno.css"` in the frontmatter of your `index.astro` file.

Our custom CSS lives there: https://github.com/stackblitz/tutorialkit/blob/main/packages/astro/src/default/styles/base.css.


As you can see there isn't much. Let us know how it goes!

**Reply by GaetanRdn** [reply]
<img width="926" alt="Capture d’écran 2024-08-05 à 17 05 18" src="https://github.com/user-attachments/assets/6890c789-3d26-4a27-b845-2bd0b3b1426f">
<img width="609" alt="Capture d’écran 2024-08-05 à 17 04 13" src="https://github.com/user-attachments/assets/7a15651f-4efe-451c-a3a9-38bd4c405e63">
<img width="635" alt="Capture d’écran 2024-08-05 à 17 04 23" src="https://github.com/user-attachments/assets/51a22dca-0515-43fd-9f33-ca00eb7c9510">
<img width="191" alt="Capture d’écran 2024-08-05 à 17 05 03" src="https://github.com/user-attachments/assets/586a7041-b3ac-43cb-9f4b-ec9a1645e53c">

I think I do what you wrote. But as you can see, I have a white background and black color.

**Reply by GaetanRdn** [reply]
If I create a theme.css file like this : https://tutorialkit.dev/reference/theming/#styling
and code ma index.astro like this :
```
---
import 'theme.css';
---

<html lang="fr" data-theme="dark">
<head>
    <meta charset="utf-8"/>
    <link rel="icon" type="image/svg+xml" href="/favicon.svg"/>
    <meta name="viewport" content="width=device-width"/>
    <meta name="generator" content={Astro.generator}>
    <title>Astro</title>
</head>
<body>
<main class="transition-theme bg-tk-elements-app-backgroundColor text-tk-elements-app-textColor">
    Hello
</main>
</body>
</html>
```

Then it works:

[
<img width="933" alt="Capture d’écran 2024-08-05 à 17 42 44" src="https://github.com/user-attachments/assets/bc744865-0055-40c0-b13d-7fd14daf9efa">
](url)

**Reply by Nemikolh** [reply]
@GaetanRdn Sorry for the late response! 

In your first example that you shared, you get the expected output when no classes are set. The reason it looks like what you shared and not like this

![image](https://github.com/user-attachments/assets/b2926656-9265-4bda-9058-84730bc6b6c5)

is because we apply a CSS reset:

https://github.com/stackblitz/tutorialkit/blob/c7bad203045b702afda3176cece645bee4d4f6e3/packages/astro/src/integrations.ts#L50

If you want it to look different you need to style it! Our default stylesheet does not affect existing elements as this provide the most flexibility and is considered good practice nowaydays.

As you noticed you can use TutorialKit classes generated by UnoCSS. You don't need to import `theme.css` or `uno.css`, this is done automatically by the UnoCSS Astro integration.

If you aren't familiar with UnoCSS, I invite you to read the online docs: https://unocss.dev/. I also find the docs from https://tailwindcss.com/ pretty useful and as the two engines are pretty close, I think you'll find them useful too.

**Reply by GaetanRdn** [reply]
Thanks for your answer 👌🏼
